numba/pythonapi.py

```diff
@@ -1201,6 +1217,31 @@ def from_native_generator(self, val, typ, env=None):
         return self.builder.call(fn,
                                  (state_size, initial_state, genfn, finalizer, env))
 
+    def nrt_adapt_native_array(self, aryty, ary):
+        if not self.context.enable_nrt:
+            raise Exception("Require NRT")
+        intty = ir.IntType(32)
+        fnty = Type.function(self.pyobj, [self.voidptr, intty, intty])
+        fn = self._get_function(fnty, name="NRT_adapt_native_array")
+        fn.args[0].add_attribute(lc.ATTR_NO_CAPTURE)
+        dtype = numpy_support.as_dtype(aryty.dtype)
+
+        ndim = self.context.get_constant(types.int32, aryty.ndim)
+        typenum = self.context.get_constant(types.int32, dtype.num)
+
+        aryptr = cgutils.alloca_once_value(self.builder, ary)
+        return self.builder.call(fn, [self.builder.bitcast(aryptr,
+                                                           self.voidptr),
+                                      ndim, typenum])
+
+    def nrt_array_adaptor(self, ary, ptr):
```

Comment:
The difference between "nrt_array_adaptor" and "nrt_adapt_native_array" is not obvious at first sight :-)